android/app/src/main/java/app/organicmaps/widget/menu/NavMenu.java

```diff
@@ -222,7 +227,8 @@ private void updateSpeedView(@NonNull RoutingInfo info)
     else
       mSpeedValue.setText(speedAndUnits.first);
 
-    if (info.speedLimitMps > 0.0 && last.getSpeed() > info.speedLimitMps)
+    final double speedLimitThresholdMps = StringUtils.nativeLocalizedSpeedToMps(SPEED_LIMIT_THRESHOLD);
```

Comment:
Calling JNI every time is not cheap. Would it be possible to include the necessary flag (or tri-state int like LAWFUL/HIT_LIMIT/HIT_THRESHOLD) in RoutingInfo struct from the C++ core and use/share it on all platforms?